### DISCUSSION TITLE
stdin in r2 command, new internal pipe syntax proposal

### DISCUSSION THREAD

**Discussion by trufae** [radareorg/radare2#19969] [Ideas]
Opening a discussion for https://github.com/radareorg/radare2/issues/14004 and closing the ticket.

The idea is to be able to write something like this:

```
pz~[0] || axt
```

the idea is that we want to be able to use pipes ini r2 commands, but r2 commands dont get anything from stdin, so instead of using babckticks and @@ iterators we need a way to express it in a pipe-form, similiar to how `xargs` work in unix

In that case that would work the same way as this:

```
axt @@=`pz~[0]`
```

But as long as || acts as `stdin` we may probably want to handle stdin pipes in more commands like `w ` for writing strings passed as stdin, those expressions could be groupped inside iterators @@.

```
izqq || w+
```

Maybe this ticket should be moved into the discussions tab

**Comment by alex-bender** [comment]
Actually a note about 'complicated' syntax is understandable.
What would that make easier?

**Comment by gogo2464** [comment]
Why not rely on the external OS pipe? Like powershell or bash?

**Reply by trufae** [reply]
Because that will require creating new processes which is something not portable. You cant do this on ios or wasm for example.
